### PR TITLE
Refactor/chat screen

### PR DESCRIPTION
**Pull request by swmd**
> Please check you are not duplicating an existing pull request...

#### Changes
* Added snapshot test of ChatNationsScreen
* Added snapshot test of ChatScreen
* Updated readme
* Implement flow of ChatScreen
* Implement eslint of ChatScreen
* Implement flow of ChatNationsScreen
* Implement eslint of ChatNationsScreen
* Implement flow of ChatNationsListScreen
* Implement eslint of ChatNationsListScreen

#### Verification
* all the tests are passed successfully.

### CODE REVIEW

**Comment by seland** [comment]
@swmd Please fix tests failed on semaphore CI.

File: __tests__/src/utils/chat.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import createGiftedChatMessageObject from '../../../src/utils/chat';
```
Comment (seland) [inline __tests__/src/utils/chat.js]: Please fix ESLint issues for that file.

File: __tests__/src/screens/ChatNationsScreen/ChatNationsListScreen/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
```
Comment (seland) [inline __tests__/src/screens/ChatNationsScreen/ChatNationsListScreen/index.js]: Please fix ESLint issues for that file.

File: __tests__/src/screens/ChatScreen/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
```
Comment (seland) [inline __tests__/src/screens/ChatScreen/index.js]: Please fix ESLint issues for that file.

File: src/screens/ChatNationsScreen/index.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
-import React, { Component } from 'react';
```
Comment (seland) [inline src/screens/ChatNationsScreen/index.js]: Please fix ESLint issues for that file.

File: src/utils/chat.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/* eslint no-underscore-dangle: 0 */
```
Comment (seland) [inline src/utils/chat.js]: Please remove this since _id was allowed by .eslintrc file.

Comment (seland) [inline src/utils/chat.js]: Please add `// @flow` for that file.

File: src/screens/ChatNationsScreen/index.js [hunk 6]
```diff
@@ -8,10 +9,37 @@ import { screen } from '../../global/Screens';
 import { resolveNation } from '../../utils/nations';
 import NavigatorComponent from '../../components/common/NavigatorComponent';
 import { startNationCreation } from '../../actions/modifyNation';
+import type { NationType } from '../../services/database/schemata';
 
+type Props = {
+  /**
+   * @desc React Native Navigation navigator object.
+   */
+  navigator: any,
```
Comment (seland) [inline src/screens/ChatNationsScreen/index.js]: Please use Navigator type from `src/types/ReactNativeNavigation.js`. Feel free to add more fields/methods there if something is missing.

File: src/screens/ChatNationsScreen/ChatNationsListScreen/index.js [hunk 7]
```diff
@@ -13,61 +14,81 @@ import NationListHeader from '../../../components/common/NationListHeader';
...
+  /**
+   * @desc List of all nations
+   */
+  nations: Array<NationType>,
+  /**
+   * @desc List of nations that the current user has joined to.
+   */
+  myNations: Array<NationType>,
+  /**
+   * @desc Function to be called when an item is selected from the list
```
Comment (seland) [inline src/screens/ChatNationsScreen/ChatNationsListScreen/index.js]: Please add params description.

File: __tests__/src/utils/chat.js (hunk 1 above)
Comment (swmd) [reply __tests__/src/utils/chat.js]: Does this file has ESLint issue?

Comment (seland) [reply __tests__/src/utils/chat.js]: Yes, please run `./node_modules/.bin/eslint __tests__/src/utils/chat.js` in terminal from root project folder to check.

Comment (seland) [reply __tests__/src/utils/chat.js]: You can also use `./node_modules/.bin/eslint __tests__/src/utils/chat.js --fix` to apply automatic fixes.

**Pull request merged**